File: backend/src/domain/entities/Notification.ts
import {
  Entity,
  PrimaryGeneratedColumn,
  Column,
  CreateDateColumn,
} from 'typeorm';

@Entity('notifications')
export class Notification {
  @PrimaryGeneratedColumn()
  id: number;

  @Column({ name: 'member_id' })
  memberId: number;

  @Column({ type: 'varchar', nullable: true })
  title: string;

  @Column({ type: 'text', nullable: true })
  message: string;

  @Column({ type: 'varchar', name: 'target_url', nullable: true })
  targetUrl: string;

  @Column({ type: 'boolean', name: 'is_read', nullable: true })
  isRead: boolean;

  @CreateDateColumn({ type: 'timestamp', name: 'created_at' })
  createdAt: Date;
}

File: backend_new_structure/common/src/main/java/com/service/common/entity/Notification.java
package com.service.common.entity;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.Id;
import org.springframework.data.relational.core.mapping.Column;
import org.springframework.data.relational.core.mapping.Table;

import java.time.LocalDateTime;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Table("notifications")
public class Notification {

    @Id
    private Long id;

    @Column("member_id")
    private Long memberId;

    private String title;

    private String message;

    @Column("target_url")
    private String targetUrl;

    @Column("is_read")
    private Boolean isRead;

    @CreatedDate
    @Column("created_at")
    private LocalDateTime createdAt;
}
